### PR TITLE
allow setting a timeout on script actions

### PR DESCRIPTION
**Pull request by bparees**
fixes https://github.com/openshift/origin/issues/14765

### CODE REVIEW

**Comment by bparees** [comment]
@stevekuznetsov ptal and then point me to where these script actions are created so i can go update some of them to pass real timeout values.  (guessing at least some of them are going to need more than an hour).

**Comment by bparees** [comment]
@stevekuznetsov to make this less risky, i've made the default 6 hours.  we can just go enforce lower timeouts as we find good places to introduce them.

File: sjb/actions/script.py [hunk 1]
```diff
@@ -43,5 +43,6 @@ def generate_build_steps(self):
             command=_SCRIPT_ACTION_TEMPLATE.render(
                 repository=self.repository,
                 command=self.script,
+                timeout=60*60*6 if self.timeout is None
```
Comment (stevekuznetsov) [inline sjb/actions/script.py]: Add `timeout=3600` as a param to the `__init__` for `ScriptAction`

**Comment by stevekuznetsov** [comment]
Thanks for the work @bparees -- actions are labeled `script:` or `host_script:` and look like [this](https://github.com/openshift/aos-cd-jobs/blob/master/sjb/config/test_cases/test_branch_origin_check.yml#L5)

**Comment by bparees** [comment]
@stevekuznetsov i found them, added a second commit w/ a few first proposals.

**Comment by bparees** [comment]
@stevekuznetsov updated.  if there are other low hanging fruit you can think of for me to slap timeouts on, let me know.

**Comment by bparees** [comment]
@stevekuznetsov test job running here:
https://ci.openshift.redhat.com/jenkins/view/All/job/zzz_test_branch_origin_extended_conformance_install_update/1/

any other comments?

**Comment by stevekuznetsov** [comment]
Once the job passes I'll merge and you can push up the changes.

**Comment by bparees** [comment]
@stevekuznetsov job passed:
https://ci.openshift.redhat.com/jenkins/view/All/job/zzz_test_branch_origin_extended_conformance_install_update/1/

**Pull request merged**